tools/test-documentation

```diff
@@ -1,92 +1,104 @@
-#!/usr/bin/env bash
-set -e
-
-color_message() {
-    local color_code="$1" message="$2"
-    printf '\e[%sm%s\e[0m\n' "$color_code" "$message" >&2
-}
-
-loglevel=()
-
-usage() {
-    cat <<EOF
-usage:
-  --help, -h                   show this help message and exit
-  --loglevel=LEVEL, -L LEVEL   log level (default: ERROR)
-  --skip-check-links           skip checking of links
-  --skip-external-links        skip checking of external links
-EOF
-}
-
-args="$(getopt -o hL: --long help,loglevel:,skip-check-links,skip-external-links -- "$@")" \
-    || {
-        usage >&2
-        exit 1
-    }
-eval "set -- $args"
-while true; do
-    case "$1" in
-        -h | --help)
-            usage
-            exit 0
-            ;;
-        -L | --loglevel)
-            loglevel=("$1" "$2")
-            shift 2
-            ;;
-        --skip-check-links)
-            skip_check_links=1
-            shift
-            ;;
-        --skip-external-links)
-            skip_external_links=1
-            shift
-            ;;
-        --)
-            shift
-            break
-            ;;
-        *) exit 1 ;;
-    esac
-done
-
-cd "$(dirname "$0")"/../docs
+#!/usr/bin/env python3
+import argparse
+import os
+import subprocess
+import sys
+from typing import Sequence
+
+ZULIP_PATH = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# check for the venv
+from lib import sanity_check
+
+sanity_check.check_venv(__file__)
+
+os.chdir(ZULIP_PATH)
+sys.path.insert(0, ZULIP_PATH)
+
+
+def color_message(color_code: str, message: str) -> None:
+    print("\033[0;{}m{}\033[0m".format(color_code, message), file=sys.stderr)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-L",
+    "--loglevel",
+    dest="loglevel",
+    nargs=1,
+    help="Log level (default: ERROR)",
+)
+parser.add_argument(
+    "--skip-check-links",
+    dest="skip_check_links",
+    action="store_true",
+    help="Skip checking of links.",
+)
+parser.add_argument(
+    "--skip-external-links",
+    dest="skip_external_link_check",
+    # default = false
+    action="store_true",
+    help="Skip checking of external links.",
+)
+options = parser.parse_args()
+# loglevel should be used as an extra command line option
+# in the form of ["-L", "LOGLEVEL"]
+# if loglevel is defined, insert "-L" ahead of loglevel
+if options.loglevel:
+    options.loglevel.insert(0, "-L")
+# if loglevel is undefined, make loglevel an empty list
+else:
+    options.loglevel = []
+
+os.chdir("docs")
 
 # collapse_navigation is set to False in conf.py to improve sidebar navigation for users.
 # However, we must change its value to True before we begin testing links.
 # Otherwise, sphinx would generate a large number of links we don't need to test.
 # The crawler would take a very long time to finish and TravisCI would fail as a result.
```

Comment:
Now we use Github Actions instead of TravisCI.

Reply:
Does that mean I need to remove Lines 56-63?

Reply:
No, I guess replace TravisCI on line 59 with Github Actions.

Reply:
I just did that.